src/sentry/plugins/base/manager.py

```diff
@@ -117,6 +118,16 @@ def install_plugin(self, plugin_cls):
             name=plugin_cls.get_full_name(), version=plugin_cls.version)
         return reg
 
+    def install_step_templates(self, plugin):
+        """
+        Installs all the step tempalates found in the plugin.
+        """
+        logger.info("Installing step templates found in plugin class '{}'".format(
+            plugin.get_name_and_version()))
+
+        for step_template_cls in plugin.get_step_templates():
+            self._app.workbatches.register_step_template(step_template_cls)
```

Comment:
I think another way to approach this is to have it in-memory only. That relates to not having models for the steps. I think there are good arguments for both approaches - currently we're doing a blend of both. I hope the diagrams will be able to shed light on how it's all today and then we have a better ground to decide which way we should go (pros/cons for each approach).

Reply:
Ah, yes. This code was also something I temporarily put up in order to make it work for now. In fact, the Step extensible model (which I forgot to remove at code where the previous comment were), is actually never registered into db (the code differ from other sorts of registrations). However, in the end I think it would be sound to have an automatic validation that there are no two steps with the same name, which can easily be checked by putting the definitions into the db.